### DISCUSSION TITLE
Media Directory Improvements

### DISCUSSION THREAD

**Discussion by kicker334** [mediacms-io/mediacms#625] [Feature Requests]
This is a feature request to somehow work the media directory differently. I already have a folder full of photos and sub folders for my "albums". Importing all of them (multiple GB) is quite a lot of work and I'm using a NFS share for my photos. I also have my phone set up to sync all new photos to this directory, so, this would require me to upload them nightly to this application. 

I don't want to duplicate photos and lose album structures. Obviously if this is the only tool I want to use forever, it makes sense to incorporate in this structure, but changing all the file names to a random string of characters doesn't allow me to go backwards if this project ever goes unsupported. 

Linking via WebDAV or through Nextcloud, Owncloud, or any of those other services could be possible. Pointing to a root directory and crawling the directory and generating thumbnails elsewhere could be possible. 

I'm not sure of a solution here, especially since I don't know the structure of the program, but this is just a feature request. 

To me, this is one of the best interfaces I've seen for replacing Google Photos.

**Comment by mgogoulos** [comment]
Hi @kicker334 , this is interesting and I want to understand more your scenario and the solutions you propose. 

We are thinking of ways to make image functionality more useful, eg 

1. adding a grid view
2. navigating through left/right arrows
3. ways to keep the hierarchy of folders during upload
